blockchain/chain_sync/src/sync.rs

```diff
@@ -137,6 +137,137 @@ where
         self.worker_state.clone()
     }
 
+    async fn handle_network_event(
```

Comment:
out of curiosity, why was this moved into a function, just to clean up or was there a functional need?

Reply:
The direct reason I did this was that rust-analyzer isn't functioning properly inside the `select!` macro. But I think this probably makes the code more readable regardless